### ISSUE TITLE
`prefer-named-capture-group` should not warn for optional groups or groups for alternations

### ISSUE THREAD

**Issue by gurpreetatwal** [eslint/eslint#12525]
**What rule do you want to change?**
`prefer-named-capture-group`

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior or a new option that allows selective disabling of this rule for those two cases

**Please provide some example code that this change will affect:**

```js
const optional = /^\d+(\.\d+)?$/u;
const alternation = /^(The|a) cat jumped$/u;
```

**What does the rule currently do for this code?**
```
  3:18  error  Capture group '(\.\d+)' should be converted to a named or non-capturing group  prefer-named-capture-group
  4:21  error  Capture group '(The|a)' should be converted to a named or non-capturing group  prefer-named-capture-group
```
**What will the rule do after it's changed?**
No warnings

**Are you willing to submit a pull request to implement this change?**
I can attempt, but don't have much familiarity with the codebase.

**Comment by mdjermanovic** [comment]
Hi @gurpreetatwal, thanks for the enhancement proposal.

I believe it's best practice to use non-capturing groups where there is no intention to capture the match (I guess that's the case here), regardless of this rule.

**Comment by gurpreetatwal** [comment]
Hi @mdjermanovic, I really appreciate the quick reply! What you said makes a lot of sense haha. I completely forgot about non-capturing groups. (have been too focused on upgrading the eslint version from 3.3. 0 to 6.6.0 on fairly large code base 😪)

I'm fine closing this issue with the resolution of just using non-capturing groups, it'll keep the eslint code simpler and enforce a best practice. What do you think?

**Comment by mdjermanovic** [comment]
> I'm fine closing this issue with the resolution of just using non-capturing groups, it'll keep the eslint code simpler and enforce a best practice. What do you think?

I agree :-)

**Comment by g-plane** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
